File: Raspberry/MeshNodeApplication.php
<?php
require_once("MeshNodeApplicationOperationCode.php");

interface MeshNodeApplication {
    /**
     * Returns the id of the application
     * @return numeric Application's identification
     */
    public function getId();
    
    /**
     * Returns the name of the application
     * @return String Applications's name
     */
    public function getName();
    
    /**
     * Returns the available OP-Codes of the application
     * @return Array of MeshNodeApplicationOperationCode
     */
    public function getOperationCodes();
    
    /**
     * Constructs a message with the specified OP-Code.
     * @param MeshNodeApplicationOperationCode 
     *        <strong>or corresponding constant</strong>
     * @return String - The message with added OP-Code.
     */
    public function constructMessage($OPCode, $message);
}

File: Android/ExampleApp/app/src/main/java/simpleblemeshnetwork/exampleapp/network/application/MeshNodeApplication.java
package simpleblemeshnetwork.exampleapp.network.application;

import java.util.List;

/**
 * @author Sebastian Blomberg
 */
public interface MeshNodeApplication {
    /**
     * Returns a <strong>unique</strong> identifier for this application
     * @return ID
     */
    public byte getId();

    /**
     * Returns the name of this application
     * @return name
     */
    public String getName();

    /**
     * Returns a textual description of this application.
     * Should include the purpose.
     * @return description
     */
    public String getDescription();

    /**
     * Returns the available operation codes (OP-Codes) of this application
     * @return
     */
    public ApplicationOperationCode[] getOperationCodes();

    /**
     * Constructs an application message with specified OP-Code to be processed
     * by this application on a node.
     * @param opCode
     * @param data
     * @return constructed message
     */
    public byte[] constructMessage(ApplicationOperationCode opCode, byte[] data);
}
